File: content-solutions/backtracking/coin-flips.js
function solution(n) {
  const result = [];
  const path = [];
  function pick() {
    if (path.length === n) {
      result.push(path.join(""));
      return;
    }
    for (const face of ["H", "T"]) {
      path.push(face);
      pick();
      path.pop();
    }
  }
  pick();
  return result;
}

File: content-solutions/backtracking/coin-flips.py
def solution(n):
    result = []
    path = []

    def pick():
        if len(path) == n:
            result.append("".join(path))
            return
        for face in "HT":
            path.append(face)
            pick()
            path.pop()

    pick()
    return result
